src/utils/sidebar.js

```diff
@@ -142,21 +135,36 @@ function normalizeSidebar({
   return currentResult
 }
 
+function findChildWithSource(item) {
+  return item.source ? item : findChildWithSource(item.children[0])
+}
+
+/*
+ * Exports
+ */
+
+// Runs at module load time
 const normalizedSidebar = normalizeSidebar({
   data: sidebar,
   parentPath: ''
 })
 
-// Exports
-
+/**
+ * Finds `path` in sidebar struct
+ * @param {*} path
+ * @uses `normalizedSidebar`
+ */
 function getItemByPath(path) {
   const normalizedPath = path.replace(/\/$/, '')
   const isRoot = normalizedPath === PATH_ROOT.slice(0, -1)
   const item = isRoot
     ? normalizedSidebar[0]
     : findItem(normalizedSidebar, normalizedPath)
 
-  return item && findChildWithSource(item)
+  if (!item) return false
+
+  // TODO: Refactor this recursive fn into a loop inside `getItemByPath`
```

Comment:
no TODOs in the code, if you feel this should be fixed create an issue/ticket with a context, explanation, priority, etc.

Reply:
Sure! Don't worry, I know. See my previous https://github.com/iterative/dvc.org/pull/891#issuecomment-570833631:

> Still draft because I need to clean up a bunch of the comments I added, and maybe migrate some to other issues.